modules/lib/lib-content/src/main/resources/lib/xp/content.ts

```diff
@@ -18,6 +18,29 @@ declare global {
 
 import type {ByteSource, Content, PublishInfo} from '@enonic-types/core';
 
+const isString = (value: unknown): value is string => value instanceof String || typeof value === 'string';
+
+const isNumber = (value: unknown): value is number => typeof value === 'number' && isFinite(value);
```

Comment:
Add `value instanceof Number` check like you did with strings to prevent `isNumber(new Number(1))` from failing:
```ts
(value instanceof Number || typeof value === 'number') && isFinite(value);
```